tests/test_env.py

```diff
@@ -11,7 +11,7 @@
 def test_isolated_environment(mocker):
     with build.env.IsolatedEnvironment() as env:
         if os.name != 'nt':
-            assert os.environ['PATH'] == os.path.join(env._path, 'bin')
+            assert os.environ['PATH'].split(os.pathsep)[-1] == os.path.join(env._path, 'bin')
```

Comment:
This test should be altered to validate that the path entries before the last are exactly what they were when invoked. Otherwise if the code above deletes all original paths and just adds the scripts at the end thing still pass 🤔